emcc.py

```diff
@@ -3243,9 +3243,10 @@ def phase_embind_emit_tsd(options, in_wasm, wasm_target, memfile, js_syms):
   # import names.
   settings.MINIFY_WASM_IMPORTED_MODULES = False
   setup_environment_settings()
-  # Replace embind with the TypeScript generation version.
-  embind_index = settings.JS_LIBRARIES.index('embind/embind.js')
-  settings.JS_LIBRARIES[embind_index] = 'embind/embind_ts.js'
+  # Ensure embind is included at least once.
+  settings.JS_LIBRARIES.index('embind/embind.js')
+  # Replace embind with the TypeScript generation version (there could be multiple).
+  settings.JS_LIBRARIES = ['embind/embind_ts.js' if s == 'embind/embind.js' else s for s in settings.JS_LIBRARIES]
```

Comment:
Perhaps we should instead make sure that `JS_LIBRARIES` contains only unique elements?   Probably a good idea anyway.

Reply:
This is already handled on the JS side https://github.com/emscripten-core/emscripten/blob/6ce6186dd93eedce7c983c680f864a5f638b7e9c/src/modules.js#L199-L200 Do we want to do it in python too?

Reply:
Maybe?  Seems like it might make sense to move that python, and then just assert it on the JS side?

Reply:
(We already have `dedup_list` in `emcc.py` BTW)